sympy/polys/polyoptions.py

```diff
@@ -152,7 +152,7 @@ def preprocess_options(args):
 
         preprocess_options(args)
 
-        for key, value in dict(defaults).items():
+        for key in dict(defaults).keys():
             if key in self:
                 del defaults[key]
```

Comment:
Using `.keys()` here is unnecessary. In fact it can be `for key in list(defaults):`

Reply:
I thought about that, but it's not quite identical. 
dict will do deduplication by keys, keep ordering, and then do it.
`for key, _ in set(defaults):`
is probably the closest thing?

Reply:
There shouldn't be any duplication because `defaults` should already be a dict.

Reply:
Also you can't have more than one default value.